src/hyperlight_host/src/hypervisor/kvm.rs

```diff
@@ -112,9 +125,30 @@ impl KVMDriver {
             entrypoint,
             orig_rsp: rsp_gp,
             mem_regions,
+
+            #[cfg(gdb)]
+            gdb_conn,
         })
     }
 
+    #[cfg(gdb)]
+    fn enable_gdb_debug(
+        mgr: Arc<Mutex<SandboxMemoryManager<GuestSharedMemory>>>,
+        vcpu_fd: Arc<Mutex<VcpuFd>>,
+        entrypoint: u64,
+    ) -> Result<GdbConnection> {
+        let (gdb_conn, hyp_conn) = GdbConnection::new_pair();
+
+        let mut target = HyperlightKvmSandboxTarget::new(mgr, vcpu_fd, entrypoint, hyp_conn);
+        let _ = target
+            .set_entrypoint_bp()
+            .map_err(|_| new_error!("Cannot set entrypoint breakpoint"))?;
```

Comment:
Is there useful information in the underlying error?

Reply:
It is a KVM generated error, I need to think about how to propagate the errors